Feature/filter records by event time (#8)
fivetran/dbt_amplitude_source: README.md

    amplitude_schema: your_schema_name 
```

## Step 4: Configure event date range

> Because of the typical volume of event data, you may want to limit this package's models to work with a more recent date range. Therefore we have added the functionality to filter records via the `date_range_start` and `date_range_end` variables within the `stg_amplitude__event` model. The default date range starts at '2020-01-01' and ends one month past the current day, but you may configure it in your root `dbt_project.yml` file:

Reviewer: ```suggestion
Because of the typical volume of event data, you may want to limit this package's models to work with a more recent date range. Therefore we have added the functionality to filter records via the `amplitude__date_range_start` and `amplitude__date_range_end` variables within the `stg_amplitude__event` model. The default date range starts at '2020-01-01' and ends one month past the current day, but you may configure it in your root `dbt_project.yml` file:
```

    amplitude_schema: your_schema_name 
```

> ## Step 4: Change event date range

Reviewer: ```suggestion
## Step 4: Configure event date range
```

> Because of the typical volume of event data, you may want to limit this package's models to work with a recent date range. Therefore we have added the functionality to filter records via the `date_range_start` and `date_range_end` variables. TThe default date range starts at '2020-01-01' and ends one day past the current day, but you may configure it in your root `dbt_project.yml` file:

Reviewer: ```suggestion
Because of the typical volume of event data, you may want to limit this package's models to work with a more recent date range. Therefore we have added the functionality to filter records via the `date_range_start` and `date_range_end` variables within the `stg_amplitude__event` model. The default date range starts at '2020-01-01' and ends one day past the current day, but you may configure it in your root `dbt_project.yml` file:
```

Reviewer: You mention it ends one day past the current. However, the model states that it is month past the current. Do we use a different default somewhere else in the package?

Reviewer: Let's make this clarification in the transform README as well. If we are using only one day past the current date for the date_spine, let's be sure to clarify the difference with the staging model.

Author: You're right, forgot to update docs to say month instead

```yml
vars:

>     amplitude:
>       date_range_start: '2022-01-01' # your start date here
>       date_range_end: '2022-12-01' # your end date here

Reviewer: Let's make these global now that the variables are used in the source **and** the transform package.
```suggestion
    date_range_start: '2022-01-01' # your start date here
    date_range_end: '2022-12-01' # your end date here
```

<details><summary>Expand to view details</summary>
<br>

### Change event date range
Because of the typical volume of event data, you may want to limit this package's models to work with a recent date range. Therefore we have added the functionality to filter records via the `date_range_start` variable. The default is '2020-01-01' but you may configure it in your root `dbt_project.yml` file:

```yml
vars:
    amplitude:
      date_range_start: '2022-01-01' # your start date here

> ```

Reviewer: I actually think this should be it's own step and not an optional config. Mainly because this will actively filter out the customers data. They should be aware of this config when setting up the package. Can we move this to the step following the database and schema variable declaration?

I am thinking a step 4 and then the additional configs will be 5 and so on.

Author: Makes sense to me, updated
